### PR TITLE
Refactor encountering with handling of all errors

### PR DESCRIPTION
**Pull request by friscoMad**
## Description
Supersedes #2078 and #2007 
I tried to handle all situations to avoid getting lv30 accounts blocked and adding encounter result handling.  Right now when there where exceptions the encounter will fail and the account will no be released properly, even the whole GMO parse will be discarded, this no longer happens.

I did not add auto captcha handling as that needs to be done globaly, currently it is only working on GMOs
I intentionally left hLvl accounts with lvl < 30 blocked forever after logging the error.
Increased the log levels for encouters to be in line with gmos and other requests.

## Motivation and Context
#2078 was abandoned and with #2026 there were some changes in the results, that lead to me to review that encounter code and trying to fix as many situations as possible.

## How Has This Been Tested?

Small instance with only 1 lvl30 account, tried to trigger a encounter error without luck as it seems to work without issues even if teleporting.
But tested a lot of hashing errors.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by AEtHeLsYn** [comment]
While this PR is great, and probably solves #2006, it would be even better combined with PR #2059 to avoid encountering with a shadowbanned lvl30 account. Anyway great job, i'll give it a try.

**Comment by RamiroWincor** [comment]
Yup #2006 should be fixed with this mostly, it is not retried but the account is not locked forever.
I am not sure what happens when you try to encounter a poke that is shadowbanned, if it doesn't work it will be logged but the account will go back to the pool, so you will be losing some encounters.
Anyway pls check and report back as I don't usually have -enc active and just got 1 account to do my tests.

**Comment by friscoMad** [comment]
Yup #2006 should be fixed with this mostly, it is not retried but the account is not locked forever.
I am not sure what happens when you try to encounter a poke that is shadowbanned, if it doesn't work it will be logged but the account will go back to the pool, so you will be losing some encounters.
Anyway pls check and report back as I don't usually have -enc active and just got 1 account to do my tests.

**Comment by SiteEffect** [comment]
@friscoMad It used to send response code 8 ``encounter_blocked_by_anticheat`` but I doubt, that this is still the case.

File: pogom/models.py [hunk 1]
```diff
@@ -2485,6 +2345,127 @@ def parse_map(args, map_dict, step_location, db_update_queue, wh_update_queue,
...
+        time.sleep(args.encounter_delay)
+
+        # If we didn't get an account, we can't encounter.
+        if not hlvl_account:
+            log.error('No L30 accounts are available, please' +
+                      ' consider adding more. Skipping encounter.')
+            return False
+
+        # Logging.
+        log.info('Encountering Pokémon ID %s with account %s at %s, %s.',
```
Comment (AEtHeLsYn) [inline pogom/models.py]: Pokemon instead Pokémon

**Comment by friscoMad** [comment]
Added the missing bits of #2007

File: pogom/models.py [hunk 2]
```diff
@@ -2485,6 +2342,127 @@ def parse_map(args, map_dict, step_location, db_update_queue, wh_update_queue,
...
+        # If not args.no_api_store is enabled, we need to
+        # re-use an old API object if it's stored and we're
+        # using an account from the AccountSet.
+        if not args.no_api_store and using_accountset:
+            hlvl_api = hlvl_account.get('api', None)
+
+        # Make new API for this account if we're not using an
+        # API that's already logged in.
+        if not hlvl_api:
+            hlvl_api = setup_api(args, status, account)
```
Comment (AEtHeLsYn) [inline pogom/models.py]: Why do we setup the hlvl_api with regular account instead of using hlvl account?

Comment (friscoMad) [inline pogom/models.py]: good call fixing it

**Pull request merged**